tests/console/verify_systemd_timesync.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that systemd timer is used for time synchronization.
+#          - Check if configuration file exists and contains expected values.
+#          - Check the message logs for "One time synchronization" occurance.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+
+    record_info("Check configuration", "Check if file /etc/systemd/system/yast-timesync.timer exists and has expected configuration.");
+    assert_script_run('ls /etc/systemd/system/yast-timesync.timer');
+    my $conf_file = script_output("cat /etc/systemd/system/yast-timesync.timer");
+    $conf_file =~ /OnUnitActiveSec=(?<interval>(\S+))min/;
+    my $expected_interval = $test_data->{profile}->{'ntp-client'}->{ntp_sync};
+    assert_equals($expected_interval, $+{interval}, "The interval in yast-timesync configuration file is not the expected one.");
+
+    record_info("Check sync", "Check if the use of systemd timers can be spotted in the journal logs as expected.");
+# The first "One time synchronization" is not happening immediately after boot. In order to make sure that first sync has already happened, there is a comparison between OnBootSec and uptime minutes
+    $conf_file =~ /OnBootSec=(?<first_sync>(\S+))min/;
+    my $uptime = script_output("uptime | cut -c19,20");
+    if ($uptime < $+{first_sync}) {
+        sleep(($+{first_sync} - $uptime) * 60);
+    }
+    elsif ($uptime == $+{first_sync}) {
+        sleep 30;
```

Comment:
regarding your findings, it seems among triggers of the service the time is a little random, so unless we have clear answer from yast developers, I would go for excluding this smart check as well, otherwise, if in some run is 31 seconds we could have some test failing and tricky to know why...
So, for this case we could implement a loop where we check every 10 seconds if it is in the logs with some reasonable maximum, so we don't waste time, we just (1) echo message + (2) check logs and (3) sleep for 10 seconds and for no more than OnBootSec * 60. I think that would be easier to understand than to have some magic numbers. wdyt?

Reply:
I did some changes. Please, check when you have time.